src/ServiceControl.Monitoring/Retries/RetriesStore.cs

```diff
@@ -1,7 +1,28 @@
 ﻿namespace ServiceControl.Monitoring
 {
+    using System;
     using Infrastructure;
-    public class RetriesStore : VariableHistoryIntervalStore
+    using Messaging;
+
+    public class RetriesStore : IProvideBreakdownBy<EndpointInstanceId>, IProvideBreakdownBy<EndpointMessageType>
     {
+        VariableHistoryIntervalStore<EndpointInstanceId> byInstance { get; } = new VariableHistoryIntervalStore<EndpointInstanceId>();
+        VariableHistoryIntervalStore<EndpointMessageType> byMessageType { get; } = new VariableHistoryIntervalStore<EndpointMessageType>();
+
+        public IntervalsStore<EndpointInstanceId>.IntervalsBreakdown[] GetIntervals(HistoryPeriod period, DateTime now)
+        {
+            return byInstance.GetIntervals(period, now);
+        }
+
+        IntervalsStore<EndpointMessageType>.IntervalsBreakdown[] IProvideBreakdownBy<EndpointMessageType>.GetIntervals(HistoryPeriod period, DateTime now)
+        {
+            return byMessageType.GetIntervals(period, now);
+        }
+
+        public void Store(RawMessage.Entry[] entries, EndpointInstanceId instanceId, EndpointMessageType messageType)
+        {
+            byInstance.Store(instanceId, entries);
+            byMessageType.Store(messageType, entries);
```

Comment:
Is this future proofing? At the moment we don't tag retries right?

Reply:
Actually we do. We tag any `duration` or `signal`. This means that every thing we measure now falls into one of these buckets.